js/app/warehouse.js

```diff
@@ -32,7 +34,7 @@ var Warehouse = new Knowledge.Class({
                     return current_dir[file_name][module_path];
                 }
                 return current_dir[file_name];
-            }, imports.app.modules);
+            }, this._import_path);
```

Comment:
Not related to this PR, but why do we have this reduce loop anyway? It implies that there should be more than one dot possible. Do we want that?

Reply:
I asked the same, but I don't see how it would be a problem either, note that this also support zero dots too (which I think is fine).